app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -174,13 +179,9 @@ class TransactionStateManager extends EventEmitter {
     @param [note] {string} - a note about the update for history
   */
   updateTx (txMeta, note) {
-    // validate txParams
+    // normalize and validate txParams if present
     if (txMeta.txParams) {
-      if (typeof txMeta.txParams.data === 'undefined') {
-        delete txMeta.txParams.data
-      }
-      txMeta.txParams = normalizeTxParams(txMeta.txParams, false)
-      this.validateTxParams(txMeta.txParams)
+      this.normalizeAndValidateTxParams(txMeta.txParams)
```

Comment:
It looks like this change is resulting in the normalization step being effectively skipped.

The `normalizeTxParams` function doesn't mutate the `txMeta.txParams` parameter - it creates a new object with the normalized parameters and returns it. So in this case, `txMeta.txParams` is getting normalized and then validated inside `normalizeAndValidateTxParams`, but these normalized params are never returned, so `txMeta.txParams` remains un-normalized.

You could preserve the previous behavior by returning `txParams` from `normalizeAndValidateTxParams, then setting `txMeta.txParams` to that return value here.

Reply:
Good catch

Reply:
You are right. My Bad. 
I think it would be good to add a test to have a check if txParams are being normalized while adding or updating. Probably in another PR because this PR is not related to normalization.

Reply:
Sure, sounds good. Thanks!